File: src/app/model/dto/ProjectUpdateDto.ts
import {RemunerationUpdateDto} from "./RemunerationUpdateDto";

export class ProjectUpdateDto {
  cup: number;
  budget: number;
  overhead: number;
  state: string;
  pnrr: boolean;
  remunerations: RemunerationUpdateDto[];

  constructor(data: any) {
    this.cup = data.cup;
    this.budget = data.budget;
    this.overhead = data.overhead;
    this.state = data.state;
    this.pnrr = data.pnrr;
    this.remunerations = data.remunerations;
  }
}

File: src/main/java/org/aldo/api/data/dto/ProjectUpdateDto.java
package org.aldo.api.data.dto;

import jakarta.validation.constraints.*;
import lombok.Data;
import org.aldo.api.security.ValidProjectId;

import java.math.BigDecimal;
import java.util.List;

@Data
public class ProjectUpdateDto {
    @NotNull
    @ValidProjectId
    private Long cup;
    @NotNull
    private BigDecimal budget;
    @NotNull
    @Min(0)
    @Max(100)
    private Integer overhead;
    @NotBlank
    @Size(min = 3, max = 40)
    private String state;
    @NotNull
    private Boolean pnrr;
    @NotEmpty
    private List<RemunerationUpdateDto> remunerations;
}
